source/proximity_list/proximitylist_api.cpp

```diff
@@ -149,13 +152,14 @@ void find(navitia::PbCreator& pb_creator,
             // we sort the proximity list on the indexes
             auto comp = [](idx_coord a, idx_coord b) { return a.first < b.first; };
             std::sort(list.begin(), list.end(), comp);
+            size_t added = 0;
 
             for (auto idx : indexes) {
                 auto it = std::lower_bound(list.begin(), list.end(), std::make_pair(idx, nt::GeographicalCoord()),
                                            comp);  // we create a mock pair and only compare the first elt
                 if (it != list.end() && (*it).first == idx) {
                     final_list.push_back(*it);
-                    if (final_list.size() == list.size()) {
+                    if (final_list.size() == list.size() || ++added == count) {
```

Comment:
I don't understand that change. Is it a debug leftover?

Reply:
No.
without this, if the filter is not empty, the search_count will be -1, then it'll return all objects that it found..

Reply:
to be honest, there is something i don't understand neither in the pagination, it seems to me that count is used a 'nb of object per page' not 'a total number of objects to be returned'...

Reply:
OK, my bad, i totally misunderstood this code...